### PR TITLE
spec: namespace annotations defined in image spec

### LINKED ISSUE #600: spec: namespace image-spec labels with "org.opencontainers.image"
When labels from images land elsewhere through propagation, it is hard to tell their source. Let's add `.image` to annotation prefix to make it clear when a label is part of an image. 

Please see https://github.com/opencontainers/image-spec/pull/492#discussion_r105034418 for the original issue.

cc @cyphar @vbatts

### PR DESCRIPTION
**Pull request by stevvooe**
Signed-off-by: Stephen J Day <stephen.day@docker.com>

Closes #600

### CODE REVIEW

**Comment by stevvooe** [comment]
@opencontainers/image-spec-maintainers PTAL

File: annotations.md [hunk 1]
```diff
@@ -9,16 +9,16 @@ Annotations MUST be a key-value map where both the key and value MUST be strings
 While the value MUST be present, it MAY be an empty string.
 Keys MUST be unique within this map, and best practice is to namespace the keys.
 Keys SHOULD be named using a reverse domain notation - e.g. `com.example.myKey`.
-Keys using the `org.opencontainers` namespace are reserved and MUST NOT be used by other specifications and extensions.
+Keys using the `org.opencontainers.image` namespace are reserved and MUST NOT be used by other specifications and extensions.
```
Comment (vbatts) [inline annotations.md]: i'm not opposed, but wonder how to handle the reservation of `org.opencontainers` prefix as well?

Comment (stevvooe) [inline annotations.md]: How about something like this:

> `org.opencontainers` is reserved for keys defined in Open Container Initiative (OCI) specifications and MUST NOT be used by other specifications and extensions. 
> Keys using the `org.opencontainers.image` namespace are reserved for use in the OCI Image Specification and MUST NOT be used by other specifications and extensions, including other OCI specifications.

Comment (jonboulle) [inline annotations.md]: SGTM

**Comment by jonboulle** [comment]
Looks good overall, with +1 to vbatts comment and your suggested solution

**Comment by stevvooe** [comment]
@vbatts @jonboulle Fixed. PTAL.

**Comment by jonboulle** [comment]
LGTM, thanks!

**Comment by vbatts** [comment]
LGTM

**Pull request merged**